File: visa-dev-samplecode-master/vdp-php/tests/vta/ManageNotificationsTest.php
<?php

namespace Vdp;

class ManageNotificationsTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$this->visaAPIClient = new VisaAPIClient;
		$this->notificationSubscriptionRequest = json_encode ([
                         "contactType"=> $this->conf['VDP'] ['vtaNotificationContactType'],
                         "contactValue"=> "john@visa.com",
                         "emailFormat"=> "None",
                         "last4"=> $this->conf['VDP'] ['vtaCreateCustomerLastFour'],
                         "phoneCountryCode"=> "en-us",
                         "platform"=> "None",
                         "preferredLanguageCode"=> $this->conf['VDP'] ['vtaPreferredLanguageCode'],
                         "serviceOffering"=> "WelcomeMessage",
                         "serviceOfferingSubType"=> "WelcomeMessage",
                         "substitutions"=> json_decode('{}')
                      ]);
	}
	
	public function testNotificationSubscription() {
		$baseUrl = "vta/";
		$resourcePath = "v3/communities/".$this->conf['VDP'] ['vtaCommunityCode']."/portfolios/".$this->conf['VDP'] ['vtaPortfolioNumber']."/customers/".$this->conf['VDP'] ['vtaCustomerId']."/notifications";
		$statusCode = $this->visaAPIClient->doMutualAuthCall( 'post', $baseUrl.$resourcePath, 'Notification Subscriptions Test', $this->notificationSubscriptionRequest, array("ServiceId: ".$this->conf['VDP'] ['vtaServiceId'] ));
		$this->assertEquals($statusCode, "201");
	}
}

File: vdp-c-sharp/vdp-c-sharp/VDPApiTest/vta/ManageNotificationsTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Configuration;
using System.Collections.Generic;

namespace Vdp
{
    [TestClass]
    public class ManageNotificationsTest
    {
        string notificationSubscriptionRequest;
        private VisaAPIClient visaAPIClient;

        public ManageNotificationsTest()
        {
            visaAPIClient = new VisaAPIClient();
            notificationSubscriptionRequest = "{"
                        + "\"contactType\":  \"" + ConfigurationManager.AppSettings["vtaNotificationContactType"] + "\","
                        + "\"contactValue\":  \"john@visa.com\","
                        + "\"emailFormat\": \"None\","
                        + "\"last4\":  \"" + ConfigurationManager.AppSettings["vtaCreateCustomerLastFour"] + "\","
                        + "\"phoneCountryCode\": \"en-us\","
                        + "\"platform\": \"None\","
                        + "\"preferredLanguageCode\":  \"" + ConfigurationManager.AppSettings["vtaPreferredLanguageCode"] + "\","
                        + "\"serviceOffering\":  \"WelcomeMessage\","
                        + "\"serviceOfferingSubType\":  \"WelcomeMessage\","
                        + "\"substitutions\": {}"
                     + "}";
        }

        [TestMethod]
        public void TestNotificationSubscription()
        {
            string baseUri = "vta/";
            string resourcePath = "v3/communities/" + ConfigurationManager.AppSettings["vtaCommunityCode"]
                + "/portfolios/" + ConfigurationManager.AppSettings["vtaPortfolioNumber"]
                + "/customers/" + ConfigurationManager.AppSettings["vtaCustomerId"] + "/notifications";
            Dictionary<string, string> headers = new Dictionary<string, string>();
            headers.Add("ServiceId", ConfigurationManager.AppSettings["vtaServiceId"]);
            string status = visaAPIClient.DoMutualAuthCall(baseUri + resourcePath, "POST", "Notification Subscriptions Test", notificationSubscriptionRequest, headers);
            Assert.AreEqual(status, "Created");
        }
    }
}
